src/instance.rs

```diff
@@ -2058,6 +2101,7 @@ where
     T: PyClass,
 {
     fn from(pyref: PyRef<'a, T>) -> Self {
+        // SAFETY: PyRef is a borrowed reference to a valid object
```

Comment:
I guess more specifically `as_ptr` returns a borrowed pointer. The `PyRef` itself owns a reference.